### ISSUE TITLE
Running in offline mode with l1c levels and specifying masking options leads to error: 

### ISSUE THREAD

**Issue by lbusett** [ranghetti/sen2r#197]
```r         
sen2r(  gui = FALSE,
            online = FALSE,
            s2_levels = "l1c",
            step_atmcorr = "no",
            extent = file.path(example_dir, "scalve.kml"),
            extent_name = "Scalve",
            extent_as_mask = TRUE,
            timewindow = as.Date("2016-10-06"),
            list_prods = "TOA",
            list_rgb = c("RGB432T", "RGB843T"),
            mask_type = "cloud_medium_proba",
            path_l1c = file.path(safe_dir, "L1C"),
            path_l2a = file.path(safe_dir, "L2A"),
            path_out = out_dir
        )

.....
[2019-08-26 16:29:42] Starting to edit geometry (clip, reproject, rescale).
trying to read file: /tmp/RtmpwB40NT/sen2r_4443e299a08/group4443e661f00/merged/SCL/S2A2A_20161006_022__SCL_10.vrt
<simpleError in UseMethod("st_as_sfc"): no applicable method for 'st_as_sfc' applied to an object of class "NULL">
Error in { : 
  task 1 failed - "task 1 failed - "task 1 failed - "no applicable method for 'st_as_sfc' applied to an object of class "NULL""""
```

Works ok if commenting the line ` mask_type = "cloud_medium_proba",`

**Comment by ranghetti** [comment]
Partially fixed with commits 5497c8815e5a5d17843d8d40718c875a0ae6f47b and d5453b589c77d8239292d0c0d6630583bd8aafea.

Now SCL is corectly generated if necessary L2A SAFE are present (or in online mode). If L1C are present in offline mode but corresponding L2A are not, the error will still be returned (this will be fixed).

**Comment by ranghetti** [comment]
Now, when both L1C-L2A archives are required to produce outputs (e.g. TOA + RGB432B), only "coupled" archives (for which both L1C and L2A exists / have been downloaded or generated) are considered. This to avoid errors i.e. if masked TOA are required.

A better approach would be manage in this way these situations in which L1C-L2A products "interacts" (i.e. SCL from L2A required to mask L1C-derived products), and to allow uncoupled products to be generated instead. This would eventually be done in the future (some work is needed for secondary situations).
